crates/core/src/client_events/websocket.rs

```diff
@@ -209,23 +220,47 @@ async fn websocket_commands(
     Extension(auth_token): Extension<Option<AuthToken>>,
     Extension(encoding_protoc): Extension<EncodingProtocol>,
     Extension(rs): Extension<WebSocketRequest>,
-) -> axum::response::Response {
+    Extension(attested_contracts): Extension<AttestedContractMap>,
+) -> Response {
     let on_upgrade = move |ws: WebSocket| async move {
-        tracing::debug!(protoc = ?ws.protocol(), "websocket connection established");
-        if let Err(error) = websocket_interface(rs.clone(), auth_token, encoding_protoc, ws).await {
+        // Get the data we need and immediately drop the lock
+        let auth_and_instance = if let Some(token) = auth_token.as_ref() {
+            // Debug the contents of the attested_contracts map
+            let attested_contracts_read = attested_contracts.read().unwrap();
+            let map_contents: Vec<_> = attested_contracts_read.keys().cloned().collect();
```

Comment:
not a big issue, but if you inline this inside the trace! it will be executed only when trace level is enabled;
you can wrap this two lines inside a bloc `{ ... }` and return the expresion, inside the macro